content/blog/architecture-as-code-intro/index.md

```diff
@@ -0,0 +1,307 @@
+---
+title: "Architecture as Code"
+date: 2020-03-18T13:42:35-05:00
+meta_desc: " An overview of fnfrastructure architecture and implementations with Pulumi"
+meta_image: architecture.png
+authors:
+    - sophia-parafina
+tags:
+    - virtual machines
+    - Kubernetes
+    - serverless
+    - microservices
+---
+
+Abstraction is key to building resilient systems because it encapsulates behavior and decouples code, letting each component perform its function independently. The same principles apply to infrastructure, where we want to declare behavior or state and not implementation details. As an industry, we've moved away from monolithic applications to distributed systems such as serverless, microservices, Kubernetes, and virtual machine deployments. In this article, we'll take a closer look at the characteristics of these architectures and how Pulumi can abstract the components that comprise these systems.
+
+<!--more-->
+
+## Virtual Machines
+
+Virtual machines arose from the need to use expensive hardware more efficiently. VMs are more efficient because they share the host server, decreasing the number of physical servers needed to run applications. A group of hosts can also be aggregated so that it behaves as a single virtual host with large memory and processing capabilities. Virtual machines can emulate other environments that have specific hardware requirements or capabilities that the host machine does not have, thus reducing the need for specific hardware or operating systems. Finally, virtualization provides the host machine isolation from the guest virtual machines by limiting access to the host machine through an abstraction layer. Finally, virtualization has the added advantage of providing a fine degree of control over resources available to the virtual machine.
```

Comment:
The last two sentences both start with "Finally,"

Reply:
fixed